Support of additional volumes in pod (#10099)
strimzi/strimzi-kafka-operator: CHANGELOG.md

## 0.43.0

Added support for additional volumes in CRDs:
* Users can specify volumes and volumeMounts fields in the Kafka, KafkaConnect, KafkaBridge, KafkaMirrorMaker2, EntityOperator, CruiseControl, KafkaExporter, and Zookeeper CRDs.
* Supported volume types include Secret, ConfigMap, EmptyDir, and PersistentVolumeClaims.

> * All additional mounted paths is mounted inside /mnt to ensure backwards compatibility for the evolution of kafka and this operator.

Reviewer: ```suggestion
* All additional mounted paths are mounted under `/mnt` to ensure backwards compatibility for the evolution of kafka and this operator.
```

Reviewer 2: Can we really claim this ensures backwards compatibility for the evolution of components external to strimzi (i.e. kafka). The directories used by components from other communities is surely a matter for those communities.

This limitation also provides no guarantees for the cluster operator as users can today mount anything they want to any location they want on the cluster operator.

Reviewer 3: We normally keep the CHANGELOG very simple. It should not replace the docs. So I think all you need there is the following:

```
* Added support for additional volumes in CRDs.
```

The rest can be left for the docs. And even the docs don't really need explanation why were things done as they were. That can be left for the proposal.

## 0.43.0

* Added alerts for Connectors/Tasks in failed state.
* Support for specifying additional volumes and volume mounts in Strimzi custom resources
* Additional OAuth configuration options have been added for 'oauth' authentication on the listener and the client. 
  On the listener `serverBearerTokenLocation` and `userNamePrefix` have been added. 
  On the client `accessTokenLocation`, `clientAssertion`, `clientAssertionLocation`, `clientAssertionType`, and `saslExtensions` have been added.

> 

Reviewer 3: Nit, but I think this line is not needed.

* Added alerts for Connectors/Tasks in failed state.

> 
> Added support for additional volumes in CRDs:
> * Users can specify volumes and volumeMounts fields in the Kafka, KafkaConnect, KafkaBridge, KafkaMirrorMaker2, EntityOperator, CruiseControl, KafkaExporter, and Zookeeper CRDs.
> * Supported volume types include Secret, ConfigMap, EmptyDir, and PersistentVolumeClaims.
> * All additional mounted paths are mounted under `/mnt` to ensure backwards compatibility for the evolution of kafka and this operator.
> * Example configurations and guidelines have been added to the documentation.

Reviewer 3: Let's try to keep the style of the other entries.

```suggestion
* Support for specifying additional volumes and volume mounts in Strimzi custom resources
```
